src/Compilers/VisualBasic/Portable/Symbols/Symbol.vb

```diff
@@ -762,7 +762,20 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
         Public Overloads Function [Equals](other As ISymbol) As Boolean Implements IEquatable(Of ISymbol).Equals
-            Return Me.[Equals](CObj(other))
+            Return Me.[Equals](other, SymbolEqualityComparer.Default.CompareKind)
+        End Function
+
+        Public Overloads Function Equals(other As ISymbol, equalityComparer As SymbolEqualityComparer) As Boolean Implements ISymbol.Equals
```

Comment:
>Public Overloads Function Equals [](start = 8, length = 32)

Let's make these `Private` if possible, and rename (the common approach is to prepend the interface name `ISymbol_`).